MVCapplication/src/main/webapp/WEB-INF/resources/js/analytics.js

```diff
@@ -20,3 +17,110 @@ $("#selectOtherProduct").click(function () {
     location.href = $(this).attr("href");
 });
 
+
+/**
+ * This block of code gets data from .jsp and displays charts of using and purchasing products
+ */
+
+$(document).ready( function() {
+
+    var purchasingProductDatesForChart = getArray($("#purchasingProductDatesForChart"));
+    var purchasingProductAmountsForChart = getArray($("#purchasingProductAmountsForChart"));
+    var usingProductDatesForChart = getArray($("#usingProductDatesForChart"));
+    var usingProductAmountsForChart = getArray($("#usingProductAmountsForChart"));
+    var endDate = $("#endDate").text().trim();
+    console.log(endDate);
+
+    var usingProductsChartArray = convertStringArrayToInteger(usingProductAmountsForChart);
+    var purchasingProductsChartArray = convertStringArrayToInteger(purchasingProductAmountsForChart);
+
+    if (endDate) {
+        usingProductDatesForChart.push(endDate);
+        usingProductsChartArray.push(0);
+    }
+
+    var dataForUsingProductsChart = {
+        labels: usingProductDatesForChart,
+        datasets: [
+            {
+                label: $("#usingProducts").text(),
+                borderColor: "rgba(238,28,37,0.5)",
+                backgroundColor: "rgba(245,121,109,0.3)",
+                data: usingProductsChartArray
+            }
+        ]
+    };
+
+    var purchasingProducts = $("#purchasingProducts").text();
+
+    var dataForPurchasingProductsChart = {
+        labels: purchasingProductDatesForChart,
+        datasets: [
+            {
+                label: purchasingProducts,
+                borderColor: "rgba(112,190,68,0.5)",
+                backgroundColor: "rgba(173,214,138,0.3)",
+                data: purchasingProductsChartArray
+            }
+        ]
+    };
+
+    var ctxForUsingProductsChart = $("#usingProductsChart");
+    var ctxForPurchasingProductsChart = $("#purchasingProductsChart");
+
+    var options = {
+            scales: {
+                yAxes: [{
+                    ticks: {
+                        beginAtZero: true
+                    }
+                }]
+            }
+        };
+
+    var usingProductsChart = new Chart(ctxForUsingProductsChart, {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png 'Severity: BLOCKER') "Chart" does not exist. Change its name or declare it so that its usage doesn't result in a "ReferenceError". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS3827)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "usingProductsChart" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AS1854)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the declaration of the unused 'usingProductsChart' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=javascript%3AUnusedVariable)